client/my-sites/site-settings/form-general.jsx

```diff
@@ -451,15 +444,37 @@ export class SiteSettingsFormGeneral extends Component {
 					) }
 				{ isWpcomStagingSite && (
 					<>
-						<PublicFormRadio />
+						{ this.publicFormRadio(
+							( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
+								( blogPublic === 0 && ! wpcomPublicComingSoon ) ||
+								blogPublic === 1,
```

Comment:
I agree with @sejas - I also found it to be a bit repetitive when looking at the code.

Reply:
I agree with @sejas about using JSX instead of calling it as a function. Calling a component as a function can lead to [various problems](https://stackoverflow.com/questions/69432301/react-hooks-call-component-as-function-vs-render-as-element).